rectools/model_selection/cross_validate.py

```diff
@@ -115,6 +116,8 @@ def cross_validate(  # pylint: disable=too-many-locals
         fold_dataset = _gen_2x_internal_ids_dataset(interactions_df_train, dataset.user_features, dataset.item_features)
 
         interactions_df_test = interactions.df.iloc[test_ids]  # 1x internal
+        if users is not None:
+            interactions_df_test = interactions_df_test[interactions_df_test[Columns.User].isin(users)]
```

Comment:
Here we already moved to internal ids for faster calculations. So it is necessary to convert external ids to internal. You can use `dataset.user_id_map` for this

Reply:
I've added the convert_to_internal method here